products/bash/apic-subscribe.sh

```diff
@@ -0,0 +1,246 @@
+  esac
+done
+
+if [ -z "${PRODUCT}" ]; then echo "PRODUCT must be specified using -p" ; exit 1 ; fi
+
+# If CONSUMER_ORG not set then default based on PROVIDER_ORG
+if [ -z "${CONSUMER_ORG}" ]; then CONSUMER_ORG=${PROVIDER_ORG}-corp; fi
+
+# If APP not set then default based on PRODUCT
+if [ -z "${APP}" ]; then APP=${PRODUCT}-app; fi
+
+TICK="\xE2\x9C\x85"
+CROSS="\xE2\x9D\x8C"
+
+client_id="599b7aef-8841-4ee2-88a0-84d49c4d6ff2"
+client_secret="0ea28423-e73b-47d4-b40e-ddb45c48bb0c"
+realm=provider/default-idp-2
+username=cp4i-admin
+password=engageibmAPI1
+APP_TITLE=${APP}
+CORG_OWNER_USERNAME="${PROVIDER_ORG}-corg-admin"
+CORG_OWNER_PASSWORD=engageibmAPI1
+
+PLATFORM_API_EP=$(oc get route -n $NAMESPACE ${RELEASE}-mgmt-platform-api -o jsonpath="{.spec.host}")
+[[ -z $PLATFORM_API_EP ]] && echo -e "[ERROR] ${CROSS} APIC platform api route doesn't exit" && exit 1
+$DEBUG && echo "[DEBUG] PLATFORM_API_EP=${PLATFORM_API_EP}"
+
+JQ=jq
+
+OUTPUT=""
+function handle_res() {
+  local body=$1
+  local status=$(echo ${body} | $JQ -r ".status")
+  $DEBUG && echo "[DEBUG] res body: ${body}"
+  $DEBUG && echo "[DEBUG] res status: ${status}"
+  if [[ $status == "null" ]]; then
+    OUTPUT="${body}"
+  elif [[ $status == "400" ]]; then
+    if [[ $body == *"already exists"* || $body == *"already subscribed"* ]]; then
+      OUTPUT="${body}"
+      echo "[INFO]  Resource already exists, continuing..."
+    else
+      echo -e "[ERROR] ${CROSS} Got 400 bad request"
+      exit 1
+    fi
+  elif [[ $status == "409" ]]; then
+    OUTPUT="${body}"
+    echo "[INFO]  Resource already exists, continuing..."
+  else
+    echo -e "[ERROR] ${CROSS} Request failed: ${body}..."
+    exit 1
+  fi
+}
+
+response=`curl -X POST https://$PLATFORM_API_EP/api/token \
+               -s -k -H "Content-Type: application/json" -H "Accept: application/json" \
+               -d "{ \"realm\": \"${realm}\",
+                     \"username\": \"${username}\",
+                     \"password\": \"${password}\",
+                     \"client_id\": \"${client_id}\",
+                     \"client_secret\": \"${client_secret}\",
+                     \"grant_type\": \"password\" }"`
+if [[ "$(echo ${response} | jq -r '.status')" == "401" ]]; then
+    echo "[ERROR] Failed to authenticate"
+    exit 1
+else
+    TOKEN=`echo ${response} | jq -r '.access_token'`
+fi
+
+# Get user registry url
+echo "[INFO] Getting configured catalog user registry url for ${PROVIDER_ORG}-catalog..."
+RES=$(curl -kLsS https://$PLATFORM_API_EP/api/catalogs/${PROVIDER_ORG}/$CATALOG/configured-catalog-user-registries \
+  -H "accept: application/json" \
+  -H "authorization: Bearer ${TOKEN}")
+handle_res "${RES}"
+
+USER_REGISTRY_URL=$(echo "${OUTPUT}" | $JQ -r ".results[0].user_registry_url")
+$DEBUG && echo "[DEBUG] User registry url: ${USER_REGISTRY_URL}"
+echo -e "[INFO] ${TICK} Got configured catalog user registry url for ${PROVIDER_ORG}-catalog"
+
+$DEBUG && echo "[DEBUG] username: $CORG_OWNER_USERNAME"
```

Comment:
```suggestion
$DEBUG && echo "[DEBUG] username: ${CORG_OWNER_USERNAME}"
```

Reply:
Doesn't really seem a necessary change, it works fine with the other syntax. I guess you just prefer the `{}` version?

Reply:
just would have matched with the other debug but its fine :D